src/server/storage.rs

```diff
@@ -130,11 +130,14 @@ impl ModelStorage {
                 }
             });
 
-        futures::future::join_all(futs)
+        // Flatten, collect, and sort on name
+        let mut models = futures::future::join_all(futs)
             .await
             .into_iter()
             .flatten()
-            .collect()
+            .collect::<Result<Vec<ModelSummary>>>()?;
+        models.sort_by_key(|model| model.name.to_owned());
```

Comment:
Nit: Can you make this return a borrowed string rather than cloning? With large lists of models this could become rather inefficient. If we have to return an owned key, it might be better to use sort_by_cached_key

Reply:
i may have to go a bit higher up and tweak some core data choices and implement a btreeset ipo hash set from get_model_set. which will do an initial ordering on String instead of post-hoc ordering in a vec and cloning.

Reply:
Yeah that is actually a really nice solution!